lib/3scale/backend/storage_async/pipeline.rb

```diff
@@ -23,12 +31,20 @@ def initialize
           # parameters for that command.
           # Ex: ['SET', 'some_key', 42].
           @commands = []
+
+          # Save the ID of the fiber that created the Pipeline so later we
+          # can check that this pipeline is not shared between fibers.
+          @fiber_id = Fiber.current.object_id
         end
 
         # In the async-redis lib, all the commands are run with .call:
         # client.call('GET', 'a'), client.call('SET', 'b', '1'), etc.
         # This method just accumulates the commands and their params.
         def call(*args)
+          if @fiber_id != Fiber.current.object_id
+            raise PipelineSharedBetweenFibers
+          end
```

Comment:
* No. It didn't show up in any of my profiling results.
* No. We need this because pipelines are implemented a bit differently in the upstream. It's another case where async-redis does not provide the same interface as redis-rb. We'll see in the future if we can contribute that "compatibility-layer" to async-redis.